src/main/c/NativeStatus.cpp

```diff
@@ -0,0 +1,42 @@
+//
+// src/main/c/NativeStatus.cpp - status/error-reporting for native C++ funcs
+//
+// Richard Cownie, Hail Team, 2018-04-24
+//
+#include "hail/NativeStatus.h"
+#include "hail/CommonDefs.h"
+#include "hail/NativePtr.h"
+#include <jni.h>
+
+NAMESPACE_BEGIN(hail)
+
+NATIVEMETHOD(long, NativeStatus, nativeCtorErrnoOffset)(
+  JNIEnv* env,
+  jobject thisObj
+) {
+  auto status = std::make_shared<NativeStatus>();
+  long errnoOffset = ((long)&status->errno_) - (long)status.get();
```

Comment:
This can just be `offsetof` (http://en.cppreference.com/w/cpp/types/offsetof), right?

Reply:
Not necessarily https://gist.github.com/graphitemaster/494f21190bb2c63c5516